gcp.go

```diff
@@ -180,29 +209,81 @@ func (l *gcpLogger) Errorf(ctx context.Context, format string, v ...any) {
 	l.log(ctx, logging.Error, fmt.Sprintf(format, v...))
 }
 
-func (l *gcpLogger) log(ctx context.Context, severity logging.Severity, p any) {
+// AddRequestAttribute adds an attribute (key, value) for the parent request log
+// If the key matches a reserved key, it will be prefixed with "custom_"
+// If the key already exists, its value is overwritten
+func (l *gcpLogger) AddRequestAttribute(key string, value any) {
+	if slices.Contains(l.rsvdKeys, key) {
+		key = customPrefix + key
+	}
+
 	l.mu.Lock()
-	if l.maxSeverity < severity {
-		l.maxSeverity = severity
+	defer l.mu.Unlock()
+	l.reqAttributes[key] = value
```

Comment:
It would be a weird use of the API, but someone could: `logger.Req(r).WithAttributes().Logger().AddRequestAttribute(key, value)`

For that to work, we would need to change this to root.
```suggestion
	l.root.mu.Lock()
	defer l.root.mu.Unlock()
	l.root.reqAttributes[key] = value
```

Reply:
The same thing applies to both `AWS` and `Console`